src/main/java/dota/buff/service/impl/HeroServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package dota.buff.service.impl;
+
+import com.github.wannesvr.core.model.hero.Hero;
+
+import dota.buff.model.HeroDTO;
+import dota.buff.service.ConvertService;
+import dota.buff.service.HeroService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class HeroServiceImpl implements HeroService {
+
+    private final List<HeroDTO> heroList;
+
+    @Autowired
+    public HeroServiceImpl(ConvertService convertService, List<Hero> allHeroes) {
+        this.heroList = convertService.convertHeroList(allHeroes);
+    }
+
+    @Override
+    public HeroDTO getHeroById(int heroId) {
+        if(heroList.size()==0){
+            throw new IllegalArgumentException("Hero list is empty");
```

Comment:
I think we can omit this check we have `orElse(null)` in case we didn't find the hero by id, but need to test it, wouldn't be an NPE inside stream when trying `hero.getId()`

Reply:
Wii be reworked in future